.github/workflows/pr-checklist.yaml

```diff
@@ -0,0 +1,51 @@
+# Context:
+#
+# Initially, the commit message for a merged github PR uses the
+# "squash and merge" setting, but oftentimes people write short and
+# uninformational commit messages, such as "more work", etc.
+#
+# We switched to another github setting that uses the PR title and
+# description for the commit message of a merged PR.
+#
+# At the same time, we would like to have a PR checklist to help
+# developers write better PRs.
+#
+# This checklist could be automatically added to all PR descriptions
+# with .github/pull_request_template.md, but this adds noise to
+# every commit message and requires manual work to remove before
+# merging.
+#
+# We decided to create a workflow to add the checklist as a comment
+# on the PR instead, so developers will have a checklist, and
+# the PR descriptions won't be cluttered.
+
+name: PR-checklist
+
+on:
+  pull_request:
+    types: [opened, edited]
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  add-checklist-comment:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: marocchino/sticky-pull-request-comment@v2
+        with:
+          header: pr-checklist
+          only_create: true
```

Comment:
Then I'm confused by those 2 settings; the opened/edited above and this only_create here.
Could you add comments explaining why we need those 2?